src/function/table/query_function.cpp

```diff
@@ -0,0 +1,75 @@
+#include "duckdb/parser/parser.hpp"
+#include "duckdb/common/helper.hpp"
+#include "duckdb/main/client_context.hpp"
+#include "duckdb/function/table/range.hpp"
+#include "duckdb/function/function_set.hpp"
+#include "duckdb/parser/tableref/subqueryref.hpp"
+
+namespace duckdb {
+
+static unique_ptr<SubqueryRef> ParseSubquery(const string &query, const ParserOptions &options, const string &err_msg) {
+	Parser parser(options);
+	parser.ParseQuery(query);
+	if (parser.statements.size() != 1 || parser.statements[0]->type != StatementType::SELECT_STATEMENT) {
+		throw ParserException(err_msg);
+	}
+	auto select_stmt = unique_ptr_cast<SQLStatement, SelectStatement>(std::move(parser.statements[0]));
+	return duckdb::make_uniq<SubqueryRef>(std::move(select_stmt));
+}
+
+static void UnionTablesQuery(TableFunctionBindInput &input, string &query) {
+	string by_name = (input.inputs.size() == 2 &&
+	                  (input.inputs[1].type().id() == LogicalTypeId::BOOLEAN && input.inputs[1].GetValue<bool>()))
+	                     ? "BY NAME "
+	                     : ""; // 'by_name' variable defaults to false
+	if (input.inputs[0].type().id() == LogicalTypeId::VARCHAR) {
+		query += "FROM " + input.inputs[0].ToString();
```

Comment:
Could we use `KeywordHelper::WriteOptionallyQuoted()` here to correctly escape the table name?